src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars. ", e);
+        }
+        for (Car car : carList) {
+            car.setDriverList(getDriversForCar(car.getId()));
+        }
+        return carList;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String sqlQuery = "UPDATE cars SET model = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE ;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(sqlQuery, Statement.RETURN_GENERATED_KEYS)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setObject(2, car.getManufacturer().getId());
+            updateCarStatement.setObject(3, car.getId());
+            updateCarStatement.executeUpdate();
+            ResultSet resultSetUpdateCar = updateCarStatement.getGeneratedKeys();
+            if (resultSetUpdateCar.next()) {
+                car.setId(resultSetUpdateCar.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update with param: " + car + ". ", e);
+        }
+        deleteOldRelation(car);
+        insertNewRelation(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, id);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car with id " + id, e);
+        }
+    }
+
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, c.model AS car_modal, "
+                + "m.id AS manufacturers_id, m.name AS manufacturers_name, "
+                + "m.country AS manufacturers_country "
+                + "FROM cars AS c "
+                + "JOIN cars_drivers ON cars_drivers.car_id = c.id "
+                + "JOIN manufacturers AS m ON m.id = c.manufacturer_id "
+                + "WHERE cars_drivers.driver_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsStatement = connection.prepareStatement(query)) {
+            getCarsStatement.setObject(1, driverId);
+            ResultSet resultSet = getCarsStatement.executeQuery();
+            List<Car> carsByDriver = new ArrayList<>();
+            Car car;
+            while (resultSet.next()) {
+                car = getCar(resultSet);
+                carsByDriver.add(car);
+            }
+
```

Comment:
```suggestion
       for (Car car : carsByDriver) {
            car.setDrivers(getDriversForCar(car.getId()));
        }
```
хоч ми і повертаємо список автомобілів для конкретного водія, давай повернемо авто з усіма проініціалізованими полями